### DISCUSSION TITLE
Dynamic schema customisation

### DISCUSSION THREAD

**Discussion by Razican** [diesel-rs/diesel#3372] [Ideas]
Currently, the `diesel_dynamic_schema` crate allows using tables and columns not defined on the migrations. This allows users to use schemas that might dynamically change on runtime (such as a DBMS).

Nevertheless, the only way to actually create new tables, drop tables and alter tables currently in Diesel is to directly use `sql_query()`. Dynamic database creation is implemented in [`diesel_cli`](https://github.com/diesel-rs/diesel/blob/master/diesel_cli/src/query_helper.rs), but it's not exposed. It would be nice to have an API in `diesel_dynamic_schema` to do the following:

 - Create database
 - Drop database
 - Alter database
 - Create table
 - Drop table
 - Alter table

**Comment by weiznich** [comment]
I would be open to accept a well written PR adding these implementations, as long as it contains tests + documentation. 
In addition I do not see any reason why that DSL implementations need to be `diesel_dynamic_schema` specific. It should be possible to write the in such a way that they work with the "normal" strongly typed DSL types as well.
